waypt.cc

```diff
@@ -74,6 +74,7 @@ void update_common_traits(const Waypoint* wpt)
   traits.trait_power |= wpt->power > 0;
   traits.trait_depth |= wpt->depth_has_value();
   traits.trait_temperature |= wpt->temperature_has_value();
+  traits.trait_sat |= wpt->sat > 0;
```

Comment:
sorry for the delay - this is just a start:

Since -1 is the default, and we may have 0 satellites, shouldn't this be "wpt->sat >= 0"?